openapi.yml

```diff
@@ -5619,6 +5904,84 @@ components:
       - trackers
       - updated_dt
       - uuid
+    FlawAcknowledgment:
+      type: object
+      description: FlawAcknowledgment serializer
+      properties:
+        name:
+          type: string
+          maxLength: 255
+        affiliation:
+          type: string
+          maxLength: 255
+        from_upstream:
+          type: boolean
+        flaw:
+          type: string
+          format: uuid
+        uuid:
+          type: string
+          format: uuid
+          readOnly: true
+        embargoed:
+          type: boolean
+          description: The embargoed boolean attribute is technically read-only as
+            it just indirectly modifies the ACLs but is mandatory as it controls the
+            access to the resource.
+        created_dt:
+          type: string
+          format: date-time
+          readOnly: true
+        updated_dt:
+          type: string
+          format: date-time
+          description: The updated_dt timestamp attribute is mandatory on update as
+            it is used to detect mit-air collisions.
+      required:
+      - affiliation
+      - created_dt
+      - embargoed
+      - flaw
+      - from_upstream
+      - name
+      - updated_dt
+      - uuid
```

Comment:
I may be wrong so take this just as something to check but is it correct to have `flaw` and `uuid` as required attributes when they are parts of the path? Just seems to be duplicite but I might miss something.

Reply:
Probably good to check with the bindings - FYI @JakubFrejlach

Reply:
Agree with @osoukup, both PUT and POST should not require flaw ID inside the request body since it is already provided in the path.

Reply:
Hopefully addressed in commit 3c2d2dd147380fc8924fac4b9d539f85f86f36d4

The swagger page for POST and PUT (when generated **without** commit 4f068286685d90f3de2ea222b5a84348059af6d8 ) has the right `Example Value`s.

POST:

```
{
  "name": "string",
  "affiliation": "string",
  "from_upstream": true,
  "embargoed": true
}
```

and PUT:

```
{
  "name": "string",
  "affiliation": "string",
  "from_upstream": true,
  "embargoed": true,
  "updated_dt": "2023-07-04T18:04:44.202Z"
}
```

`flaw_id` and `id` are still marked as required, but also marked as belonging to `path` as opposed to `query`.

Does that do what you need @JakubFrejlach please? Thanks!

Reply:
This is blocking other work (OSIDB-1004) and it seems the concerns have been addressed.